cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -616,10 +615,10 @@ protected:
     void chooseFeatures()
     {
         const size_t n    = nFeatures();
-        const size_t nGen = (!_par.memorySavingMode && !_maxLeafNodes && !_useConstFeatures) ? n : _nFeaturesPerNode;
+        const size_t nGen = (!_par.memorySavingMode && !_maxLeafNodes && !_par.useConstFeatures) ? n : _nFeaturesPerNode;
```

Comment:
Make sure you add this into https://github.com/oneapi-src/oneDAL/blob/master/cpp/daal/include/algorithms/decision_forest/decision_forest_training_parameter.h
and
https://github.com/oneapi-src/oneDAL/blob/master/cpp/daal/src/algorithms/dtrees/forest/df_training_parameter.cpp
otherwise it will always evaluate to false.

Reply:
Thanks! I forgot to initialize it, but now it's included everywhere